### PR TITLE
Allow to disable auto-stop for SQL Endpoints

### PR DESCRIPTION
**Pull request by alexott**

### CODE REVIEW

File: common/reflect_resource.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func StructToSchema(v interface{}, customize func(map[string]*schema.Schema) map
 }
 
 func handleOptional(typeField reflect.StructField, schema *schema.Schema) {
-	if strings.Contains(typeField.Tag.Get("json"), "omitempty") {
+	if strings.Contains(typeField.Tag.Get("json"), "omitempty") || strings.Contains(typeField.Tag.Get("tf"), "optional") {
```
Comment (nfx) [inline common/reflect_resource.go:88]: This may cause schema consistency validation errors, need to run also integration tests just to be sure

Comment (alexott) [reply common/reflect_resource.go:88]: If you afraid that something will break, then I can get back to original fix:

```
m["auto_stop_mins"].Required = false
m["auto_stop_mins"].Optional = true
```

Comment (nfx) [reply common/reflect_resource.go:88]: Required is false by default anyway. 

This is literally the same annoying case as with `num_workers`, where we have a lot of hacks around.

Comment (alexott) [reply common/reflect_resource.go:88]: No. If I remove omitempty then it’s implicitly set to Required = true during analysis of the struct… I already hit this issue :-)

Comment (alexott) [reply common/reflect_resource.go:88]: Yes. I need to check, maybe we can simplify other places with that

Comment (nfx) [reply common/reflect_resource.go:88]: there should be no struct tags for single place use

File: sqlanalytics/resource_sql_endpoint.go [hunk 2]
```diff
@@ -186,6 +186,8 @@ func ResourceSQLEndpoint() *schema.Resource {
 	s := common.StructToSchema(SQLEndpoint{}, func(
 		m map[string]*schema.Schema) map[string]*schema.Schema {
 		m["auto_stop_mins"].Default = 120
+		m["auto_stop_mins"].Required = false
+		m["auto_stop_mins"].Optional = true
```
Comment (nfx) [inline sqlanalytics/resource_sql_endpoint.go:190]: `.Optional = true` and no `omitempty` creates a schema consistency conflict.

**Review by nfx — CHANGES_REQUESTED** [review]

**Comment by nfx** [comment]
this PR is of low priority, given the percentage of resource usage across all resources.

File: sqlanalytics/resource_sql_endpoint.go (hunk 2 above)
Comment (alexott) [reply sqlanalytics/resource_sql_endpoint.go:190]: that's why it has `.Required = false`

**Pull request merged**